test/specs/wrapper/overview.spec.js

```diff
@@ -0,0 +1,336 @@
+          '    bar: [',
+          '        0: [ hey ]',
+          '    ]',
+          '}',
+          ''
+        ]
+        wrapper.overview()
+        expect(consoleOutput).to.have.ordered.members(expectedConsoleOutput)
+      })
+    })
+
+    describe('computed', () => {
+      it('prints Computed as {} when data is empty', () => {
+        const wrapper = mountingMethod({
+          template: '<div class="test" style="{ color: red }"></div>'
+        })
+        wrapper.vm.$emit('foo', 'hello', 'world')
+        wrapper.vm.$emit('foo', 'bye', 'world')
+        wrapper.vm.$emit('bar', 'hey')
+
+        const expectedConsoleOutput = [
+          '',
+          'Wrapper (Visible):',
+          '',
+          'Html:',
+          '    <div class="test"></div>',
+          '',
+          'Data: {}',
+          '',
+          'Computed: {}',
+          '',
+          'Emitted: {',
+          '    foo: [',
+          '        0: [ hello, world ],',
+          '        1: [ bye, world ]',
+          '    ],',
+          '    bar: [',
+          '        0: [ hey ]',
+          '    ]',
+          '}',
+          ''
+        ]
+        wrapper.overview()
+        expect(consoleOutput).to.have.ordered.members(expectedConsoleOutput)
+      })
+    })
+
+    describe('emitted events', () => {
+      it('prints Emitted as {} when no events have been emitted', () => {
+        const wrapper = mountingMethod({
+          template: '<div class="test" style="{ color: red }"></div>'
+        })
+
+        const expectedConsoleOutput = [
+          '',
+          'Wrapper (Visible):',
+          '',
+          'Html:',
+          '    <div class="test"></div>',
+          '',
+          'Data: {}',
+          '',
+          'Computed: {}',
+          '',
+          'Emitted: {}',
+          ''
+        ]
+        wrapper.overview()
+        expect(consoleOutput).to.have.ordered.members(expectedConsoleOutput)
+      })
+    })
+
+    describe('child components', () => {
+      it('prints children compenents HTML', () => {
+        if (mountingMethod.name === 'shallowMount') return
+        const wrapper = mountingMethod({
+          template: `<div>1<tester></tester></div>`,
+          components: {
+            tester: {
+              template: `<div class="tester">test</div>`
+            }
+          }
+        })
+
+        const expectedConsoleOutput = [
+          '',
+          'Wrapper (Visible):',
+          '',
+          'Html:',
+          '    <div>1<div class="tester">test</div>',
+          '    </div>',
+          '',
+          'Data: {}',
+          '',
+          'Computed: {}',
+          '',
+          'Emitted: {}',
+          ''
+        ]
+
+        wrapper.isVisible = () => true
+        wrapper.overview()
+        expect(consoleOutput).to.have.ordered.members(expectedConsoleOutput)
+      })
+
+      it('does not print child component data or computed', () => {
+        if (mountingMethod.name === 'shallowMount') return
```

Comment:
I might be missing something - why are we skipping `shallowMount`?

Reply:
The test checks if the HTML shown in the overview includes the child component. 

With mount child component is:
`<div class="tester">test</div>`

But with shallow mount is something like this:
`<tester-stub></tester-stub>`

I don't know if it's really necessary to test this with shallow mount, but to make the test clearer I can add a diferent expect for every mounting method. 

For example:

```javascript
const expectedConsoleOutput = [
	  '',
	  'Wrapper (Visible):',
	  '',
	  'Html:',
	  mountingMethod.name === 'shallowMount'
	    ? '    <div>1<tester-stub></tester-stub>'
	    : '    <div>1<div class="tester">test</div>',
	  '    </div>',
	  '',
	  'Data: {}',
	  '',
	  'Computed: {}',
	  '',
	  'Emitted: {}',
	  ''
	]
```

Reply:
> it ('does not print child component data or computed')

I got the test mixed up. You're talking about the one for computed and data. In this case it makes more sense, anyways the solution would be the same.

Reply:
I think it's fine to show the stubbed html for the shallow mounted components. shallowMount's output is basically useless, but I think it's better to be consistent and show it anyway. At least from my point of view, I would expect overview to show the same information for both rendering APIs.